partiql-spi/src/main/java/org/partiql/spi/value/Datum.java

```diff
@@ -774,6 +577,13 @@ static Datum struct(@NotNull Iterable<Field> values) {
         return new DatumStruct(values);
     }
 
+    @NotNull
+    static Datum ion(@NotNull String value) {
+        IonElementLoader loader = ElementLoader.createIonElementLoader();
+        AnyElement element = loader.loadSingleElement(value);
+        return new IonVariant(element);
+    }
```

Comment:
I went back-and-forth on this quite a bit, but stopped myself from setting up a whole VariantLoader which is where this eventually leads. I would suggest leaving any variant instance (i.e. "ion" or "json) out of the datum public API, rather it makes sense to have either

1.  IonVariant be public (which I know you would not like) with private constructor
2. Have the API be `Datum.variant(String encoding, String payload)` or something equivalent.

This way we don't couple any variant name/instance to Datum's public API.

Reply:
Seems I've contradicted myself (: 

https://github.com/partiql/partiql-lang-kotlin/blob/main/partiql-spi/src/main/java/org/partiql/spi/value/DatumReader.java#L31-L37

Reply:
😄 Yeah, haha, that's actually where I got the naming from. But, LMK whatever your preference is! Since the default encoding is Ion when a user writes back-ticks, I'm not opposed to keeping the method as `ion()`.